document/controller.py

```diff
@@ -0,0 +1,121 @@
+        pass
+    
+    @abstractmethod
+    def delete_document(self, doc_id: str):
+        pass
+      
+class DocumentControllerV1(ABC):
+    def __init__(self, service: DocumentService):
+        self.service = service
+        self.logger = logger.bind(service="DocumentController")
+        super().__init__()
+        
+    def fetch_documents(
+      self, 
+      id: str | None = None,
+      object_name: str | None = None,
+      created_after: str | None = None,
+      created_before: str | None = None,
+      updated_after: str | None = None,
+      updated_before: str | None = None,
+    ):
+        doc_filter = DocumentFilter(
+          id = id,
+          object_name = object_name,
+          created_after = created_after,
+          created_before = created_before,
+          updated_after = updated_after,
+          updated_before = updated_before,
+        )
+        try:
+            docs = self.service.get_documents(filter=doc_filter)
+            if docs is None:
+                return HTTPException(status_code=404)
+            return docs
+        except Exception as e:
+            self.logger.error(e.args)
+            return HTTPException(status_code=500)
+      
+    def fetch_document_by_name(self, object_name: str):
+        try:
+            doc = self.service.get_document_by_name(object_name=object_name)
+            if doc is None:
+                return HTTPException(status_code=404)
+            return doc
+        except Exception as e:
+            self.logger.error(e.args)
+            return HTTPException(status_code=500)
+      
+    def fetch_document_by_id(self, doc_id: str):
+        try:
+            doc = self.service.get_document_by_id(doc_id=doc_id)
+            if doc is None:
+                return HTTPException(status_code=404)
+            return doc
+        except Exception as e:
+            self.logger.error(e.args)
+            return HTTPException(status_code=500)
+
+    def upload_document(self, 
+      file: Annotated[UploadFile, File()], 
+      # request: Annotated[DocumentCreate, Form()]
+      type: Annotated[str, Form()],
+      object_name: Annotated[str, Form()],
+      title: Annotated[str, Form()],
+    ):
+        doc_create = DocumentCreate(
+          type=type,
+          object_name=object_name,
+          title=title,
+        )
+        
+        # TODO: S3 integration (upload file)
+        self.logger.info(file.filename)
+        
+        try:
+            self.service.create_document(request=doc_create)
+            return {"detail": "Document created successfully!"}
+        except DocumentTypeError:
+            raise HTTPException(status_code=400, detail="Document Type is not valid")
+        except DocumentTitleError:
+            raise HTTPException(status_code=400, detail="Document title is required")
+        except ObjectNameError:
+            raise HTTPException(status_code=400, detail="Object name is required")
```

Comment:
Since you've already defined the exception alongside it's message, you can just use the exception's message property directly to prevent drift and duplication like this

```suggestion
        except DocumentTypeError as exc:
            raise HTTPException(status_code=400, detail=exc.message)
        except DocumentTitleError as exc:
            raise HTTPException(status_code=400, detail=exc.message)
        except ObjectNameError as exc:
            raise HTTPException(status_code=400, detail=exc.message)
```

Reply:
brilliant move